codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -159,14 +169,16 @@ class KotlinJmespathExpressionVisitor(
                 append("if ($isNullExpr) null else ")
             }
 
-            val unSafeComparatorExpr = "compareTo(${right.identifier}) ${expression.comparator} 0"
+            val unSafeComparatorExpr = "compareTo(${right.identifier})"
```

Comment:
Also this generates code like:

```kotlin
            val unSafeComparator = if (integer == null) null else integer?.compareTo(number)
            val safeComparator = if (unSafeComparator == null) null else unSafeComparator == 0
            safeComparator == true
```

Where you're just taking the null value if it's null otherwise comparing it to the expected value. You can compare `null` directly though

```kotlin
val comparator = integer == 0
```

Another example:
```kotlin
            val lists = it.lists
            val strings = lists?.strings
            val length = strings?.length ?: 0
            val number = 0.0
            val unSafeComparator = if (length == null) null else length.compareTo(number)
            val safeComparator = if (unSafeComparator == null) null else unSafeComparator > 0
            safeComparator == true
```

it can be condensed to:

```kotlin
            val safeComparator = length > 0
```

If the left expression of the comparision is nullable it can be changed to:

```
val safeComparator = length?.let { it > 0 }
```

Which will yield `Boolean?` as the result.